crates/sui-config/src/genesis.rs

```diff
@@ -342,29 +344,53 @@ impl GenesisValidatorInfo {
         if !self.info.name.is_ascii() {
             bail!("name must be ascii");
         }
-        if self.info.name.len() > 128 {
-            bail!("name must be <= 128 bytes long");
+        if self.info.name.len() > MAX_VALIDATOR_METADATA_LENGTH {
+            bail!("name must be <= {MAX_VALIDATOR_METADATA_LENGTH} bytes long");
         }
 
         if !self.info.description.is_ascii() {
             bail!("description must be ascii");
         }
-        if self.info.description.len() > 150 {
-            bail!("description must be <= 150 bytes long");
+        if self.info.description.len() > MAX_VALIDATOR_METADATA_LENGTH {
+            bail!("description must be <= {MAX_VALIDATOR_METADATA_LENGTH} bytes long");
+        }
+
+        if self.info.image_url.len() > MAX_VALIDATOR_METADATA_LENGTH {
+            bail!("image url must be <= {MAX_VALIDATOR_METADATA_LENGTH} bytes long");
+        }
+
+        if self.info.project_url.len() > MAX_VALIDATOR_METADATA_LENGTH {
+            bail!("project url must be <= {MAX_VALIDATOR_METADATA_LENGTH} bytes long");
         }
 
         if !self.info.network_address.to_string().is_ascii() {
             bail!("network address must be ascii");
         }
+        if bcs::to_bytes(&self.info.network_address)?.len() > MAX_VALIDATOR_METADATA_LENGTH {
```

Comment:
I think we want this to be done via to_string since we encode these as strings now on-chain

Reply:
good point, I will check the inner bytes' length